lib/hawkbit/hawkbit.c

```diff
@@ -0,0 +1,1212 @@
+
+	hb_context.dl.http_content_size = 0;
+	ret = hawkbit_parse_deployment(&hawkbit_results.dep, &json_action_id,
+				       download_http, &file_size);
+	if (ret < 0) {
+		LOG_ERR("Unable to parse deployment base");
+		goto cleanup;
+	}
+
+	nvs_read(&fs, ADDRESS_ID, &device_action_id, sizeof(device_action_id));
+
+	if (device_action_id.update == (u32_t)json_action_id) {
+		LOG_INF("Preventing repeated attempt to install %d",
+			json_action_id);
+		hb_context.dl.http_content_size = 0;
+		memset(hb_context.url_buffer, 0, sizeof(hb_context.url_buffer));
+		hb_context.url_buffer_size = URL_BUFFER_SIZE;
+		snprintk(hb_context.url_buffer, hb_context.url_buffer_size,
+			 "%s/%s-%s/deploymentBase/%d/feedback",
+			 HAWKBIT_JSON_URL, CONFIG_BOARD, device_id,
+			 json_action_id);
+
+		if (!send_request(HTTP_POST, HAWKBIT_REPORT,
+				  HAWKBIT_STATUS_FINISHED_SUCCESS,
+				  HAWKBIT_STATUS_EXEC_CLOSED)) {
+			LOG_ERR("Error when querying from hawkbit");
+			hb_context.code_status = HAWKBIT_NETWORKING_ERROR;
+			goto cleanup;
+		}
+		hb_context.code_status = HAWKBIT_OK;
+		goto cleanup;
+	}
+
+	LOG_INF("Ready to install update");
+
+	hb_context.dl.http_content_size = 0;
+	memset(hb_context.url_buffer, 0, sizeof(hb_context.url_buffer));
+	hb_context.url_buffer_size = URL_BUFFER_SIZE;
+
+	snprintk(hb_context.url_buffer, hb_context.url_buffer_size, "%s",
+		 download_http);
+
+	flash_img_init(&hb_context.flash_ctx);
+
+	if (!send_request(HTTP_GET, HAWKBIT_DOWNLOAD,
+			  HAWKBIT_STATUS_FINISHED_NONE,
+			  HAWKBIT_STATUS_EXEC_NONE)) {
+		LOG_ERR("Send request failed");
+		hb_context.code_status = HAWKBIT_NETWORKING_ERROR;
+		goto cleanup;
+	}
+
+	if (boot_request_upgrade(BOOT_UPGRADE_TEST)) {
+		LOG_ERR("Download failed");
+		hb_context.code_status = HAWKBIT_DOWNLOAD_ERROR;
+	} else {
+		hb_context.code_status = HAWKBIT_UPDATE_INSTALLED;
+		hawkbit_device_acid_update(HAWKBIT_ACTION_ID_UPDATE,
+				json_action_id);
+	}
+
+	hb_context.dl.http_content_size = 0;
+
+cleanup:
+	cleanup_connection();
+
+error:
+	return hb_context.code_status;
+}
+
+static void autohandler(struct k_delayed_work *work)
+{
+	switch (hawkbit_probe()) {
+	case HAWKBIT_UNCONFIRMED_IMAGE:
+		LOG_ERR("Image is unconfirmed");
+		LOG_ERR("Rebooting to previous confirmed image.");
+		sys_reboot(SYS_REBOOT_WARM);
+		break;
+
+	case HAWKBIT_NO_UPDATE:
+		LOG_INF("No update found");
+		break;
+
+	case HAWKBIT_CANCEL_UPDATE:
+		LOG_INF("Hawkbit update Cancelled from server");
+		break;
+
+	case HAWKBIT_OK:
+		LOG_INF("Image is already updated");
+		break;
+
+	case HAWKBIT_UPDATE_INSTALLED:
+		LOG_INF("Update Installed. Please Reboot");
+		break;
+
+	case HAWKBIT_DOWNLOAD_ERROR:
+		LOG_INF("Update Failed");
+		break;
+
+	default:
+		break;
+	}
+
+	k_delayed_work_submit(work, poll_sleep);
+}
+
+void hawkbit_autohandler(void)
+{
+	static struct k_delayed_work work;
+
+	k_delayed_work_init(&work, autohandler);
+	k_delayed_work_submit(&work, K_NO_WAIT);
+}
```

Comment:
As a general note to hawkbit.c, you are mostly using LOG_INF(). Usually that use is a bit limited, depends on your use cases of course. Anyway, some of the info prints could probably be more appropriate in LOG_DBG().